pkg/river/internal/value/decode.go

```diff
@@ -397,15 +398,16 @@ func tryCapsuleConvert(from Value, into reflect.Value, intoType Type) (ok bool,
 // decodeAny is invoked by decode when into is an interface{}. We assign the
 // interface{} a known type based on the River value being decoded:
 //
-//	Null values:   nil
-//	Number values: float64, int, or uint depending on the underlying Go type
-//	               of the River value
-//	Arrays:        []interface{}
-//	Objects:       map[string]interface{}
-//	Bool:          bool
-//	String:        string
-//	Function:      Passthrough of the underlying function value
-//	Capsule:       Passthrough of the underlying capsule value
+//			Null values:           nil
+//			Number values:         float64, int64, or uint64 depending on the underlying Go type
+//		                           of the River value. If the underlying type is not a float and the
+//	                               number fits within an "int", then an "int" will be used.
```

Comment:
The alignment got a little messed up here:

```suggestion
//			Number values:         float64, int64, or uint64 depending on the underlying Go type
//			                       of the River value. If the underlying type is not a float and 
//			                       the number fits within an "int", then an "int" will be
//			                       used.
```